osidb/models.py

```diff
@@ -1808,6 +1845,76 @@ class Meta:
             "created_dt",
         )
 
+    def bzsync(self, *args, bz_api_key=None, **kwargs):
+        """
+        Bugzilla sync of the FlawComment instance and of the related Flaw instance.
+        """
+
+        self.save()
+
+        # Comments need to be synced through flaw
+        # If external_system_id is blank, BugzillaSaver posts the new comment
+        # and FlawCollector loads the new comment and updates this FlawComment
+        # instance to match bugzilla.
+        self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
+
+    def _validate_single_pending_comment(self):
+        """
+        Check that this is at most the first pending comment for the flaw.
+        This comment must not be a second pending comment for the flaw. If it
+        is, it means that API calls are not ATOMIC_REQUESTS anymore, or that
+        the temporary pending comments are not cleaned up properly, or that
+        something makes concurrent requests.
+        """
+        tmp_cmnts = FlawComment.objects.pending().filter(flaw=self.flaw)
+        if tmp_cmnts.count() > 1:
```

Comment:
I am not sure this is necessary. If the atomicity does not work, we are screwed anyway when the scheduler stops the thread before this line and lets another thread to get to the same point (after the check but before the actual save). I would just assume that the atomicity works and Django is fully responsible for it.

Otherwise we could question many other things like do the Django really stores the model to the DB after `save` and rather check it by a pure SQL select to be truly sure and we could think about other examples.

Reply:
I agree that we have to act as if Django works well enough so that we don't fall into a rabbit hole of checks. Hopefully a Django config change will effect an actually working mutually exclusive/locked transactions. Just tested the check and it doesn't ensure atomicity in concurrent situations anyway. Related to discussion https://github.com/RedHatProductSecurity/osidb/pull/209/#discussion_r1209214388 (`https://github.com/RedHatProductSecurity/osidb/pull/209/#discussion_r1209214388`)

Reply:
As I mentioned above, let us not go crazy with ensuring the mutual exclusion. Just minimize it but that should be done with already existing checks. To much complexity is IMO not worth as the risk is already low anyway.

Reply:
Resolved in https://github.com/RedHatProductSecurity/osidb/pull/209/commits/f9ebfa5badd141532da083c3d6506b796e4868b4